examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/support/EndpointFilter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *   Copyright (c) 2024 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.matter.casting.support;
+
+import java.util.List;
+
+/** Describes an Endpoint that the client wants to connect to. */
+public class EndpointFilter {
+  // Value of 0 means unspecified
+  public int productId;
+  // Value of 0 means unspecified
+  public int vendorId;
```

Comment:
In Java, use Integer (not int) and let null mean that it is unspecified.

Reply:
Changed the field types int -> Integer